src/axom/sina/core/Document.cpp

```diff
@@ -138,37 +291,63 @@ void saveDocument(Document const &document, std::string const &fileName)
   // file is in the same directory to ensure that it is part of the same
   // file system as the destination file so that the move operation is
   // atomic.
+
   std::string tmpFileName = fileName + SAVE_TMP_FILE_EXTENSION;
-  auto asJson = document.toJson();
-  std::ofstream fout {tmpFileName};
-  fout.exceptions(std::ostream::failbit | std::ostream::badbit);
-  fout << asJson;
-  fout.close();
 
-  if(rename(tmpFileName.c_str(), fileName.c_str()) != 0)
+  if (protocol == Protocol::JSON)
   {
-    std::string message {"Could not save to '"};
-    message += fileName;
-    message += "'";
-    throw std::ios::failure {message};
+      protocol_warn(".json", fileName);
+      auto asJson = document.toJson();
+      std::ofstream fout {tmpFileName};
+      fout.exceptions(std::ostream::failbit | std::ostream::badbit);
+      fout << asJson;
+      fout.close();
+  }
+  else if (protocol == Protocol::HDF5)
+  {
+      protocol_warn(".hdf5", fileName);
+      document.toHDF5(tmpFileName);
+  }
+  else
+  {
+      throw std::invalid_argument("Invalid format choice. Please enter 'json' or 'hdf5'.");
+  }
+
+  if (rename(tmpFileName.c_str(), fileName.c_str()) != 0)
+  {
+      std::string message {"Could not save to '"};
+      message += fileName;
+      message += "'";
+      throw std::ios::failure {message};
   }
 }
 
-Document loadDocument(std::string const &path)
+Document loadDocument(std::string const &path, Protocol protocol)
 {
-  return loadDocument(path, createRecordLoaderWithAllKnownTypes());
+  return loadDocument(path, createRecordLoaderWithAllKnownTypes(), protocol);
 }
 
-Document loadDocument(std::string const &path, RecordLoader const &recordLoader)
+Document loadDocument(std::string const &path, RecordLoader const &recordLoader, Protocol protocol)
 {
-  conduit::Node nodeFromJson;
-  std::ifstream file_in {path};
-  std::ostringstream file_contents;
-  file_contents << file_in.rdbuf();
-  file_in.close();
-  nodeFromJson.parse(file_contents.str(), "json");
-  return Document {nodeFromJson, recordLoader};
+    conduit::Node node;
+    
+    // Load the file depending on the protocol
+    if (protocol == Protocol::JSON) {
+        std::ifstream file_in {path};
+        std::ostringstream file_contents;
+        file_contents << file_in.rdbuf();
+        file_in.close();
+        node.parse(file_contents.str(), "json");
+        return Document {node, recordLoader};
+    } else if (protocol == Protocol::HDF5) {
+        conduit::Node modifiedNode;
+        conduit::relay::io::load(path, "hdf5", node);
+        restoreSlashes(node, modifiedNode);
+        return Document {modifiedNode, recordLoader};
+    }
```

Comment:
Similar to `saveDocument` we might consider using a `switch` statement here as well.